### PR TITLE
(#3637) Build NCIDS Feature Card Multimedia

### LINKED ISSUE #3637: User Story: Build NCIDS Feature Card Multimedia
## User Story:
**As a** content editor, 
**I want to** have the Development Solution for Drupal, 
**so that I** can create NCIDS Feature Card Multimedia components on an NCI website

**Parent Issue:** #3492 

> **ESTIMATE** TBD

### Acceptance Criteria

**TODO:** Need to discuss how files should display file and size information, or if it is even needed.

*Drupal Display:*
``` gherkin
Scenario: User can see fields for an NCIDS Feature Card Multimedia
  Given user is logged in to the CMS
    And user edits a "Home and Landing" content item
  When user adds an "NCIDS 3 Feature Card Row" to the "Contents" field
    And user adds an "NCIDS Feature Card Multimedia" to the "Feature Cards" field
  Then user will see the fields for an "NCIDS Feature Card Multimedia"
    And first field will be "Featured Media"
    And "Featured Media" field will be required
    And "Featured Media" field will be an entity reference to a media item
    And next field will be "Override Card Title"
    And "Override Card Title" field will be optional
    And "Override Card Title" field will be a text input
    And next field will be "Override Card Description"
    And "Override Card Description" field will be optional
    And "Override Card Description" field will be a text input
    And next field will be "Override Promotional Image"
    And "Override Promotional Image" field will be optional
    And "Override Promotional Image" field will be an image select
```

*Drupal Functionality:*
``` gherkin
Scenario: User can add an NCIDS Feature Card Multimedia to an NCIDS 3 Feature Card Row
  Given user is logged into the CMS
  When user creates a new "Home and Landing" content item
  Then the editing page for a "Home and Landing" content item displays
    And user fills out the following fields
      | fieldLabel                  | value                                                            | field_name                     |
      | Home and Landing Page Style | NCIDS with Title                                                 | field_page_style               |
      | Pretty URL                  | home-and-landing-page                                            | field_pretty_url               |
      | Page Title                  | Automated Test NCIDS Feature Card Multimedia                     | title                          |
      | Browser Title               | Automated Test NCIDS Feature Card Multimedia - Browser Title     | field_browser_title            |
      | Card Title                  | Automated Test NCIDS Feature Card Multimedia - Card Title        | field_card_title               |
      | Meta Description            | Automated Test NCIDS Feature Card Multimedia Meta Description    | field_page_description         |
      | Feature Card Description.   | Automated Test NCIDS Feature Card Multimedia - Feature Card Desc | field_feature_card_description |
  When user selects "Add NCIDS 3 Feature Card Row" from "Contents" dropdown
  Then "NCIDS 3 Feature Card Row" section appears
    And user fills out the following fields
      | fieldLabel | value                    | field_name              |
      | Heading    | NCIDS 3 Feature Card Row | field_container_heading |
  When user selects "Add NCIDS Feature Card Multimedia" from "Feature Cards" dropdown
  Then "NCIDS Feature Card Multimedia" section appears
    And user clicks on "Featured Media" link in the "NCIDS Feature Card Multimedia" area
    And user clicks on "Select content" button item
    And user selects "NCI at a Glance Infographic" item from the list
    And user clicks on "Select content" button to select item
    And "NCI at a Glance Infographic" had been selected
    And user fills out the following fields
      | fieldLabel                | value                                                 | field_name                        |
      | Override Card Title       | Override Card Title NCIDS Feature Card Multimedia       | field_override_card_title       |
      | Override Card Description | Override Card Description NCIDS Feature Card Multimedia | field_override_card_description |
    When user saves the content page
    Then user can see NCIDS Feature Card Multimedia on the page
      And card displays an image
      And card displays the title "Override Card Title NCIDS Feature Card Multimedia"
      And card displays the description "Override Card Description NCIDS Feature Card Multimedia"
```

```gherkin
Scenario: Analytics event for feature card clicks
Given a user is on any page of cancer.gov with feature cards
When the user clicks on a link within the feature card
Then there should be an NCIDataLayer Other event 
  And the event has the name “LP:FeatureCard:LinkClick”
  And the event has the linkname “LP:FeatureCard:LinkClick”
  And this event should have the following data:
    | key              | value|
    | location         | Body |
    | pageRows         | This should return the number of rows on the page overall |
    | pageRowIndex     | This should return the index of the row number on the page that was clicked |
    | rowItems         | This should return the total number of components in the row that is clicked|
    | rowItemIndex     | This should return the index of the component in the rows that was clicked |
    | componentType    | Feature Card |
    | componentTheme   | This should be “Light” or “Dark”. In cases where a theme option is not yet available, this match whatever the available theme color is.|
    | componentVariant | This value represents the component design options available for a component type. For Feature Cards, this value should be set to “Standard Single Link”, “Standard Multi-Link”, or “Wide Single Link”. |
    | title            | This value should capture the card title field displayed. This value should be truncated after 50 characters. |
    | linkType         | This value can be set to “internal”, “external”, or “media” based on the way the card is setup|
    | linkText         | This value will change slightly based on the linkArea. If a user clicks on the title, this should pass the text of the title value (truncate after 50). If the user clicks on the description, this should pass the text of the description value (truncate after 50). If the user clicks on the image, pass “Image”. If the user clicks on the secondary link, pass the text displayed for that link. |
    | linkArea         | For feature cards, this value should be set to “Image”, “Description”, “Title”, or “Secondary Link” based on where the user clicks. If the area cannot be determined, please pass "Undefined" |
    | totalLinks       | This should pass the total number of links shown in the card. For standard feature cards this will be 1. When multilink cards are created in the future, this will be 2. |
    | linkPosition     | This should pass the index of the link clicked in the card evaluated by counting the links top to bottom. For standard feature cards this will be 1. When multilink cards are created in the future, this will be 1 or 2 depending on the link clicked. |

```

### Resources:
* Component variant design: [NCIDS #605](https://github.com/NCIOCPL/ncids/issues/605)

---

### Prerequisites
- https://github.com/NCIOCPL/cgov-digital-platform/issues/3551
- https://github.com/NCIOCPL/cgov-digital-platform/issues/3519

### Tasks
- [ ] [Create NCIDS Feature Card Multimedia paragraph](https://github.com/NCIOCPL/cgov-digital-platform/wiki/paragraph-type-ncids-feature-card#multimedia-card)
- [x] #3690
- [x] #3691
- [ ] Create basic twig template
- [ ] Evaluate component using acceptance criteria in the Parent issue, #3492 

### Links
**Parent**: #3492 
**Siblings**: #3611, #3612, #3613

### Technical Notes
**Content Modeling:**
- [NCIDS Feature Card Multimedia](https://github.com/NCIOCPL/cgov-digital-platform/wiki/paragraph-type-ncids-feature-card#multimedia-card)
- [NCIDS 3 Feature Card Row](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Paragraph-Type-NCIDS-3-Feature-Card-Row)

**Media Link Title Display Rules**

Media items display mostly the same ways as the internal and external links, the difference is that the title contains information about the media type or file information.
All titles are either the Name of the media item, or the Override Title. 
Only the title portion of the link is clickable and responds to hover - the media type does not. 

#### Infographic

*English:* `<Infographic Title> (Infographic)`

*Spanish:* `<Infographic Title> (Infografía)`

#### Video

*English and Spanish:* `<Video Title> (Video)`

#### Files
*English and Spanish:* `<File Title> (<icon or filetype>) (Size: <unit>)`
- If the file is .html or .htm then the title displays only as `<File Title>`
- Only these icons are available for display - any other extensions show as . (and have no icon)
  - pdf
  - MS Word (.doc and .docx)
  - MS Excel (.xls and .xlsx)
  - MS Powerpoint (.ppt and .pptx)
  - mobi
  - epub
  - exe
  - txt
- For the file size, use these rules
  - Round the file size to the nearest integer.
  - Display the file size in bytes if the file size is less than 1 KB.
  - Display the file size in KB if the file size is less than 1 MB.
  - Display the file size in MB if the file size is more than 1 MB.

**Drupal image display flow:**

image

### PR DESCRIPTION
**Pull request by dev-rana-publicis**
Closes #3637
ODE: https://ncigovcdode596.prod.acquia-sites.com/

### CODE REVIEW

**Comment by bennettcc** [comment]
This has passed analytics team review on the ODE as of 7/10

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/cards/media--ncids-feature-card-mm-image.html.twig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  {{content.field_image_promotional}}
+{% else %}
+  {# Placeholder Image #}
+  {% set sourceAttribs = create_attribute({
+    'media': '(min-width: 880px)'
+  })
+  %}
+  <picture class="nci-card__image">
+    <source {{ sourceAttribs }} srcset="{{ get_placeholder_image('ncids_featured_4x3') }}">
+    <img src="{{ get_placeholder_image('ncids_featured_16x9') }}" alt="No Image Placeholder">
```
Comment (olitharp-nci) [inline docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/cards/media--ncids-feature-card-mm-image.html.twig:12]: hardcoded image alt?

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Review by bennettcc — APPROVED** [review]

**Comment by bennettcc** [comment]
Adding the product checkmark following design, IA, and analytics reviews - all passed as of 7/11

File: docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/cards/media--ncids-feature-card-mm-image.html.twig (hunk 1 above)
Comment (bryanpizzillo) [reply docroot/profiles/custom/cgov_site/themes/custom/ncids_trans/templates/cards/media--ncids-feature-card-mm-image.html.twig:12]: This is the image placeholder alt, so it should not be an issue.

**Review by olitharp-nci — APPROVED** [review]

**Comment by bennettcc** [comment]
Feds have reviewed and approved this component as of 7/20

**Review by bryanpizzillo — APPROVED** [review]

**Pull request merged**